types/src/transaction/program.rs

```diff
@@ -162,3 +185,62 @@ impl fmt::Debug for TransactionArgument {
         }
     }
 }
+
+impl CanonicalSerialize for TransactionArgument {
+    fn serialize(&self, serializer: &mut impl CanonicalSerializer) -> Result<()> {
+        match self {
+            TransactionArgument::U64(value) => {
+                serializer.encode_u32(TransactionArgument_ArgType::U64 as u32)?;
+                serializer.encode_u64(*value)?;
+            }
+            TransactionArgument::Address(address) => {
+                serializer.encode_u32(TransactionArgument_ArgType::ADDRESS as u32)?;
+                serializer.encode_struct(address)?;
+            }
+            TransactionArgument::String(string) => {
+                serializer.encode_u32(TransactionArgument_ArgType::STRING as u32)?;
+                serializer.encode_vec(string.as_bytes())?;
+            }
+            TransactionArgument::ByteArray(byte_array) => {
+                serializer.encode_u32(TransactionArgument_ArgType::BYTEARRAY as u32)?;
+                serializer.encode_struct(byte_array)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl CanonicalDeserialize for TransactionArgument {
+    fn deserialize(deserializer: &mut impl CanonicalDeserializer) -> Result<Self> {
+        let decoded_value = deserializer.decode_u32()? as i32;
+        let arg_type = TransactionArgument_ArgType::from_i32(decoded_value);
+        let arg_type: TransactionArgument_ArgType = match arg_type {
+            Some(v) => v,
```

Comment:
This can be rewritten as an `unwrap_or_else`, or an [`if let`](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html)